src/api/codeApi.ts

```diff
@@ -0,0 +1,248 @@
+import { AxiosInstance } from "axios";
+import { Api } from "coder/site/src/api/api";
+import {
+	GetInboxNotificationResponse,
+	ProvisionerJobLog,
+	ServerSentEvent,
+	Workspace,
+	WorkspaceAgent,
+} from "coder/site/src/api/typesGenerated";
+import { type WorkspaceConfiguration } from "vscode";
+import { ClientOptions } from "ws";
+import { CertificateError } from "../error";
+import { getHeaderCommand, getHeaders } from "../headers";
+import {
+	createRequestMeta,
+	logRequest,
+	logRequestError,
+	logResponse,
+} from "../logging/httpLogger";
+import { Logger } from "../logging/logger";
+import { RequestConfigWithMeta, HttpClientLogLevel } from "../logging/types";
+import { WsLogger } from "../logging/wsLogger";
+import { OneWayWebSocket } from "../websocket/oneWayWebSocket";
+import { createHttpAgent } from "./auth";
+
+const coderSessionTokenHeader = "Coder-Session-Token";
+
+type WorkspaceConfigurationProvider = () => WorkspaceConfiguration;
+
+/**
+ * Unified API class that includes both REST API methods from the base Api class
+ * and WebSocket methods for real-time functionality.
+ */
+export class CodeApi extends Api {
```

Comment:
What do you think about calling this `CoderApi`?  `CodeApi` to me feels like an API to Code rather than to Coder.

Reply:
Sweet!  `Api` is also a mix of REST and WS, btw, except we are not using the WS parts.